.gitignore

```diff
@@ -1 +1,32 @@
-.vscode/
\ No newline at end of file
+# Prerequisites
+*.d
+
+# Compiled Object files
+*.slo
+*.lo
+*.o
+*.obj
+
+# Precompiled Headers
```

Comment:
why would we need this?

Reply:
.obj files (.o files on Linux/Unix) are compiled source files, there will indeed be one for each .cpp file

Reply:
okay I don't see them when we run our commands. When are we producing .obj files?

Reply:
This comment applies to all:
1.  if you don't want these additions to the .gitignore, then reject them.
2.  I am struggling to understand why we are spending so much time on this. I think we both have better ways to spend our time then debate .gitignore options
3. All the lines that I copied from github are _possible_ files that could be included in a cpp project.  Most we haven't created so far.

Reply:
1. We need to ignore files and you are right in making this. They just need to be the files we create from make
2. I am sure we do, but its important to do this right as we are not trying to git ignore a lot of things we do not need.
3. We do not need possible files. We need to manage the ones we have.

Reply:
We clearly don't see eye to eye on this.  Do what ever you want with the .gitignore file - I do not care but it seems quite important to you.

Reply:
Ann, If I wanted to cut and paste what someone else recommends that is not doing this project I can easily do it. The thing you need to learn here is we only need to ignore the artifacts we are creating with the make file ( *.html *.bin *.exe) nothing else. If we start creating other file extensions or other environment binaries we will add then to the list as we go along. You ae making it more comlicated then it needs to be